src/renderers/shared/fiber/ReactFiberCommitWork.js

```diff
@@ -53,10 +60,22 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
     getPublicInstance,
   } = config;
 
+  function callComponentWillUnmountWithTimerInDev(current, instance) {
```

Comment:
That's cool. Would be good validate as this is something we can improve with flat bundling if it's not happening here.